src/dataMap.js

```diff
@@ -100,27 +97,32 @@ class DataMap {
    */
   createMap() {
     const schema = this.getSchema();
-    let i;
 
     if (typeof schema === 'undefined') {
       throw new Error('trying to create `columns` definition but you didn\'t provide `schema` nor `data`');
     }
 
+    const columns = this.tableMeta.columns;
+    let i;
+
     this.colToPropCache = [];
     this.propToColCache = new Map();
 
-    const columns = this.tableMeta.columns;
-
     if (columns) {
-      const maxCols = this.tableMeta.maxCols;
-      let columnsLen = Math.min(maxCols, columns.length);
+      let columnsLen = null;
```

Comment:
Maybe `0` will be better for initialization 🙂?

Reply:
Generally I'd rather initialize with `null`, so if something goes wrong in the logic, it won't fail silently with a _potentially_ valid value, but in this specific case I guess I can make it `0`.